### PR TITLE
Update EOL versions in release/8.0-staging

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

File: eng/pipelines/common/templates/pipeline-with-resources.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ resources:
 
     # We use a CentOS Stream 8 image here to test building from source on CentOS Stream 8.
```
Comment (directhex) [inline eng/pipelines/common/templates/pipeline-with-resources.yml:58]: Comment needs to match image

**Comment by richlander** [comment]
Looks like HttpClient is failing on CentOS Stream 9. 

FYI: https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/

@dotnet/ncl

**Comment by wfurt** [comment]
seems like msquic is missing in the images. I do see packages published https://packages.microsoft.com/rhel/9/prod/Packages/l/
I think the fix should come from the prerequisites repo @richlander

**Review by wfurt — APPROVED** [review]

**Comment by wfurt** [comment]
it seems like the test runs are clean @richlander

**Comment by richlander** [comment]
We still need the correct label to be applied. I don't know what that is.

**Comment by wfurt** [comment]
this is test only change. It should be OK to just merge it right @carlossanlop ?

**Comment by richlander** [comment]
I just realized I edited some files in the `common` folder. Those changes should be removed, right?

**Comment by wfurt** [comment]
I don't know. 
There is IMHO difference between `eng/pipelines/common/` and `eng/common`. First probably belongs to runtime, second belongs to arcade. @akoeplinger or @ViktorHofer would probably know for sure.

**Comment by richlander** [comment]
Ah. You may well be right. We'll wait for the experts to weigh in.

**Comment by richlander** [comment]
Can you review @carlossanlop  and/or @hoyosjs ?

**Review by carlossanlop — APPROVED** [review]
LGTM :shipit:

**Comment by richlander** [comment]
Thanks for the help everyone!

**Pull request merged**